test/reputation-system/reputation-mining-client/client-auto-functionality.js

```diff
@@ -164,6 +165,102 @@ process.env.SOLIDITY_COVERAGE
           await miningCycleComplete;
         });
 
+        it("should follow updates if it is only an oracle", async function() {
+          const oracleClient = new ReputationMinerClient({
+            loader,
+            realProviderPort,
+            minerAddress: MINER2,
+            useJsTree: true,
+            oracle: true,
+            auto: false
+            // dbPath: "./reputationStates_good.sqlite"
+          });
+
+          await oracleClient.initialise(colonyNetwork.address, startingBlockNumber);
+
+          const rootHash = await reputationMinerClient._miner.getRootHash();
+          const nNodes = await reputationMinerClient._miner.getRootHashNNodes();
+          const jrh = await reputationMinerClient._miner.justificationTree.getRootHash();
+
+          const oldHash = await colonyNetwork.getReputationRootHash();
+          const { minerAddress } = reputationMinerClient._miner;
+
+          const repCycleEthers = await reputationMinerClient._miner.getActiveRepCycle();
+          const receive12Submissions = new Promise(function(resolve, reject) {
+            repCycleEthers.on("ReputationRootHashSubmitted", async (_miner, _hash, _nNodes, _jrh, _entryIndex, event) => {
+              const nSubmissions = await repCycle.getNSubmissionsForHash(rootHash, nNodes, jrh);
+              if (nSubmissions.toNumber() === 12) {
+                // Check the reputation cycle submission matches our miner
+                // Validate the miner address in submission is correct
+                const lastSubmitter = await repCycle.getSubmissionUser(rootHash, nNodes, jrh, 11);
+                expect(lastSubmitter).to.equal(minerAddress);
+
+                // Validate the root hash matches what the miner submitted
+                const submission = await repCycle.getReputationHashSubmission(minerAddress);
+                expect(submission.proposedNewRootHash).to.equal(rootHash);
+
+                event.removeListener();
+                resolve();
+              } else {
+                await mineBlock();
+              }
+            });
+
+            // After 30s, we throw a timeout error
+            setTimeout(() => {
+              reject(new Error("ERROR: timeout while waiting for 12 hash submissions"));
+            }, 30000);
+          });
```

Comment:
This section doesn't involve the oracleClient at all and looks a bit boilerplate-y -- any way this can be pulled out into a helper function? Or let me know if I'm mistaken.